File: examples/sdk/python/basic_conversation.py
"""
Basic Conversation Example
This example shows how to start and manage a basic sales conversation
"""

import asyncio
import os
from ngx_voice_sales import NGXVoiceSales

# Initialize the client
client = NGXVoiceSales(
    api_key=os.environ.get('NGX_API_KEY')
)


async def main():
    try:
        # 1. Start a new conversation
        print('Starting conversation...')
        
        conversation = await client.conversations.create(
            customer_info={
                "name": "Carlos Rodriguez",
                "business_type": "gym",
                "business_size": "medium",
                "location": "Barcelona",
                "current_challenges": ["member retention", "engagement"]
            },
            initial_message="Hola, tengo un gimnasio con 500 miembros y estoy perdiendo clientes"
        )
        
        print(f'\n🤖 Agent: {conversation.response}')
        print(f'📊 Conversation ID: {conversation.id}')
        
        # 2. Continue the conversation
        print('\n👤 Customer: ¿Cuánto cuesta NGX?')
        
        response1 = await client.conversations.send_message(
            conversation_id=conversation.id,
            message="¿Cuánto cuesta NGX?",
            include_analytics=True
        )
        
        print(f'🤖 Agent: {response1.message}')
        print(f'📈 Sentiment: {response1.analytics.sentiment}')
        
        # 3. Handle objection
        print('\n👤 Customer: Me parece caro para mi presupuesto')
        
        response2 = await client.conversations.send_message(
            conversation_id=conversation.id,
            message="Me parece caro para mi presupuesto"
        )
        
        print(f'🤖 Agent: {response2.message}')
        
        # 4. Get ROI calculation
        print('\n💰 Calculating ROI...')
        
        roi = await client.roi.calculate(
            conversation_id=conversation.id,
            business_metrics={
                "current_members": 500,
                "monthly_churn": 0.15,
                "average_ticket": 89,
                "current_monthly_revenue": 44500
            }
        )
        
        print('📊 ROI Analysis:')
        print(f'  - Monthly Savings: €{roi.monthly_savings}')
        print(f'  - ROI: {roi.roi_percentage}%')
        print(f'  - Payback Period: {roi.payback_months} months')
        
        # 5. Get conversation summary
        summary = await client.analytics.get_conversation(conversation.id)
        
        print('\n📋 Conversation Summary:')
        print(f'  - Conversion Probability: {summary.conversion_probability * 100:.1f}%')
        print(f'  - Customer Sentiment: {summary.sentiment}')
        print(f'  - Detected Needs: {", ".join(summary.detected_needs)}')
        print(f'  - Recommended Tier: {summary.recommended_tier}')
        
    except Exception as e:
        print(f'❌ Error: {str(e)}')
        
        if hasattr(e, 'code') and e.code == 'RATE_LIMIT_EXCEEDED':
            print(f'⏱️  Retry after {e.retry_after} seconds')


# Run the example
if __name__ == "__main__":
    asyncio.run(main())

File: examples/sdk/javascript/basic-conversation.js
/**
 * Basic Conversation Example
 * This example shows how to start and manage a basic sales conversation
 */

import { NGXVoiceSales } from '@ngx/voice-sales-sdk';

// Initialize the client
const client = new NGXVoiceSales({
  apiKey: process.env.NGX_API_KEY
});

async function main() {
  try {
    // 1. Start a new conversation
    console.log('Starting conversation...');
    
    const conversation = await client.conversations.create({
      customerInfo: {
        name: "Carlos Rodriguez",
        businessType: "gym",
        businessSize: "medium",
        location: "Barcelona",
        currentChallenges: ["member retention", "engagement"]
      },
      initialMessage: "Hola, tengo un gimnasio con 500 miembros y estoy perdiendo clientes"
    });
    
    console.log('\n🤖 Agent:', conversation.response);
    console.log('📊 Conversation ID:', conversation.id);
    
    // 2. Continue the conversation
    console.log('\n👤 Customer: ¿Cuánto cuesta NGX?');
    
    const response1 = await client.conversations.sendMessage({
      conversationId: conversation.id,
      message: "¿Cuánto cuesta NGX?",
      includeAnalytics: true
    });
    
    console.log('🤖 Agent:', response1.message);
    console.log('📈 Sentiment:', response1.analytics.sentiment);
    
    // 3. Handle objection
    console.log('\n👤 Customer: Me parece caro para mi presupuesto');
    
    const response2 = await client.conversations.sendMessage({
      conversationId: conversation.id,
      message: "Me parece caro para mi presupuesto"
    });
    
    console.log('🤖 Agent:', response2.message);
    
    // 4. Get ROI calculation
    console.log('\n💰 Calculating ROI...');
    
    const roi = await client.roi.calculate({
      conversationId: conversation.id,
      businessMetrics: {
        currentMembers: 500,
        monthlyChurn: 0.15,
        averageTicket: 89,
        currentMonthlyRevenue: 44500
      }
    });
    
    console.log('📊 ROI Analysis:');
    console.log(`  - Monthly Savings: €${roi.monthlySavings}`);
    console.log(`  - ROI: ${roi.roiPercentage}%`);
    console.log(`  - Payback Period: ${roi.paybackMonths} months`);
    
    // 5. Get conversation summary
    const summary = await client.analytics.getConversation(conversation.id);
    
    console.log('\n📋 Conversation Summary:');
    console.log(`  - Conversion Probability: ${(summary.conversionProbability * 100).toFixed(1)}%`);
    console.log(`  - Customer Sentiment: ${summary.sentiment}`);
    console.log(`  - Detected Needs: ${summary.detectedNeeds.join(', ')}`);
    console.log(`  - Recommended Tier: ${summary.recommendedTier}`);
    
  } catch (error) {
    console.error('❌ Error:', error.message);
    
    if (error.code === 'RATE_LIMIT_EXCEEDED') {
      console.log(`⏱️  Retry after ${error.retryAfter} seconds`);
    }
  }
}

// Run the example
main();